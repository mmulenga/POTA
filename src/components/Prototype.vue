<template>

    <!-- Columns start at 50% wide on mobile and bump up to 33.3% wide on desktop -->
    <div class="row">
        <!-- This is the left side view -->
        <div class="col-6 col-md-4">
            <!-- TODO: actually make this do something - when menu button is clicked it should
            show/hide the options of home/glossary/help. If home is clicked, it should take you
            back to the main question page, if glossary is clicked it should take to you a page
            all terminology, if help is clicked, it will direct you to a page for help -->
            <h4 class="alert-heading"></h4>
                <button class="btn btn-primary" type="button" id="menu_btn">Menu</button>
                <hr>
                <div class="nav flex-column nav-pills py-5" id="v-pills-tab"
                role="tablist" aria-orientation="vertical">
                    <a class="nav-link active" id="testing_btn"
                        data-toggle="pill" href="#v-pills-home"
                        role="tab" aria-controls="v-pills-home"
                        aria-selected="true">Testing</a>
                    <a class="nav-link" id="help_btn"
                        data-toggle="pill" href="#v-pills-profile"
                        role="tab" aria-controls="v-pills-profile"
                        aria-selected="false">Help</a>
                    <a class="nav-link" id="glossary_btn"
                        data-toggle="pill" href="#v-pills-settings"
                        role="tab" aria-controls="v-pills-settings"
                        aria-selected="false">Glossary</a>
                </div>
        </div>
        <!-- This is the mid side view -->
        <div class="col-6 col-md-4">To Do the Mid side UI</div>
        <!-- This is the right side view (UI) -->
        <div class="col-6 col-md-4">
            <div class="alert alert-success" role="alert">
                <h4 class="alert-heading">Recommend Test:</h4>
                <red-text>
                    Showing what test should be done
                </red-text>
                <hr>
                <red-text>
                    Showing what test should be done
                </red-text>
                <hr>
                <red-text>
                    Showing what test should be done
                </red-text>
            </div>
            <br/>
            <br/>
            <br/>
            <br/>
            <br/>
            <br/>
            <br/>
            <br/>
            <div class="alert alert-success" role="alert">
                <h4 class="alert-heading">Status of Patient:</h4>
                <red-text>
                    Showing patient's status
                </red-text>
                <br/>
                <hr>
                <button class="btn btn-primary" type="button" id="reset_btn">Restore</button>
            </div>
        </div>
    </div>
</template>

<script>

export default {
  name: 'Prototype',
  data() {
    return {
      framework_name: 'VueJS',
    };
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style>
    red-text{
        color:red;
    }
</style>
