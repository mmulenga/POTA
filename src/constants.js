/**
 * This file contains all string constants for the project
 * Use this instead of hardcoding string values
 * HTML and one-time-use messages are an acceptable exception
 */

/* Conditions */
export const Condition = {
  /* Cardiovascular Diseases */
  AtrialFib: 'Atrial fibrillation / History of irregular heart beat',
  Defib: 'Defibrillator / Pacemaker',
  CAD: 'Coronary artery disease / Cardiac stent',
  CerebralDisease: 'History of cerebral vascular disease / TIA / Stroke',
  VHD: 'Valvular heart disease / Valve Replacement',
  HeartFail: 'Heart failure',
  PVD: 'Peripheral Vascular disease (PVD)',
  /* Pulmonary Diseases */
  PulmonaryVascular: 'History of pulmonary vascular disease',
  PulmDisease: 'Severe COPD, Home Oxygen, Pulmonary HTN',
  /* Other Diseases */
  Risk: 'Age ≥ 69 years old OR ≥ 2 risk factors (HTN, DM, OSA, CKD, BMI > 35)',
  Bleeding: 'Bleeding disorders',
  Anemia: 'History of anemia in past 12 months or ongoing blood loss',
  KidneyDisease: 'Chronic kidney disease / on dialysis',
  Diabetes: 'Diabetes',
  Malignancy: 'Present malignancy / surgery for malignancy',
  Hepatic: 'Hepatic disease',
  Adrenal: 'History of adrenal disease',
  Pituitary: 'History of pituitary disease',
  Endocrine: 'History of major systemic endocrine disease',
  Thyroid: 'Known thyroid disease or on thyroid supplementation',
  Digoxin: 'Digoxin',
  Lithium: 'Lithium',
  Diuretics: 'Diuretics',
  ACEI: 'ACE-I',
  ARB: 'ARB',
  NSAIDS: 'NSAIDS',
  Anticoagulant: 'Anticoagulants',
  Antiplatelet: 'Antiplatelet (ASA excluded)',
  Steroid: 'Systemic steroid use within 6 months',
};

/* Medical Examinations */
export const Exam = {
  GnS: 'Group and Screen',
  ECG: 'ECG',
  CBC: 'CBC',
  RenPanel: 'Renal Panel (Creat + Lytes)',
  PTTINR: 'PTT/INR',
  LFT: 'LFT',
  Gluc: 'Gluc',
  HbA1C: 'HbA1C',
  CXR: 'CXR',
  TSH: 'TSH',
};

/* Glossary Entries */
export const Glossary = {
  AtrialFib: 'No extra information.',
  Defib: 'No extra information.',
  CAD: 'No extra information.',
  CardiacStent: 'No extra information.',
  CerebralDisease: 'No extra information.',
  PulmonaryVascular: 'No extra information.',
  VHD: 'No extra information.',
  HeartFail: 'No extra information.',
  PVD: 'No extra information.',
  /* Pulmonary Diseases */
  PulmDisease: '• Spirometric values AND symptomatic assessment should be sought. \n' +
               '• Post Bronchodilator FEV1: 30 - 50% of predicted; FEV1/FVC < 0.70 \n' +
               '• History of exacerbations (including prior hospitalizations) should be recorded.\n' +
               '• Worsening air flow limitation, greater SOB, reduced exercise capacity, having an ' +
               'impact on quality of life, and ability to complete activities of daily living should be present. \n',
  /* Other Diseases */
  Risk: 'No extra information.',
  Bleeding: '• Hemophiliac. \n' +
            '• History of extensive investigation due to a bleeding episode. \n' +
            '• History of extensive bleeding needing transfusion in previous surgeries. \n' +
            '• Bruising easily more than usual. \n',
  Anemia: 'No extra information.',
  ActiveBleeding: 'No extra information.',
  KidneyDisease: 'No extra information.',
  Renal: '(AKI). \n' +
         '• Chronic kidney disease / on dialysis. \n' +
         '• Diabetes. \n' +
         '• Heart failure. \n' +
         '• Age ≥ 65. \n' +
         '• Liver disease. \n' +
         '• Use of drugs with nephrotoxic potential in the perioperative period (in particular NSAIDs after surgery). \n',
  Diabetes: 'No extra information.',
  Malignancy: 'No extra information.',
  Hepatic: '• Alcohol abuse (i.e., female: 14 drinks / week; male: 21 drinks / week). \n' +
           '• Jaundice due to liver disease. \n' +
           '• All forms of hepatitis. \n' +
           '• Cirrhosis; Hepatic cancer. \n',
  Adrenal: 'No extra information.',
  Pituitary: '• Acromegaly. \n' +
             '• Cushing’s disease. \n' +
             '• Sheehan syndrome. \n' +
             '• Growth hormone deficiency. \n' +
             '• Hyper/hypo pituitarism. \n',
  Endocrine: '(Excluding Diabetes) \n' +
             '• Parathyroid disease \n' +
             '• Addison’s disease \n' +
             '• Cushing’s syndrome \n' +
             '• Graves’ disease \n',
  Thyroid: 'No extra information.',
  /* Medications */
  Digoxin: 'No extra information.',
  Lithium: 'No extra information.',
  Diuretics: 'Furosemide; Hydrochlorothiazide; Acetazolamide; Spironolactone; Chlorthalidone',
  ACEI: 'Captopril; Enalapril; Ramipril; Trandolapril; Perindopril',
  ARB: 'Valsartan; Candesartan; Irbesartan; Telmisartan; Losartan; Olmesartan',
  NSAIDS: 'No extra information.',
  Anticoagulant: 'Warfarin; Heparins; Dabigatran; Apixaban; Rivaroxaban',
  Antiplatelet: 'Clopidogrel; Ticagrelor; Prasugrel; Aspirin/Dipyridamole (Aggrenox)',
  Steroid: 'Prednisone; Prednisolone; Methylprednisolone; Hydrocortisone; Dexamethasone; Oral Budesonide',
};
