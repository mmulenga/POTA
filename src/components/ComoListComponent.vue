<template>
  <div class="container">
    <div class="row">
      <div class="col-12">
        <ul class="list-group half">
          <li class="list-group-item list-group-item-dark"> Cardiovascular Diseases</li>
          <li class="list-group-item list-group-item-action"
          v-for="(item, index) in cardioDiseases"
          :key="item.comorbidity"
          v-on:click="aggregateConditions(cardioDiseases, index)">
            <div class="row">
              <div class="col-2">
                <div class="form-check">
                  <input class="form-check-input" type="checkbox" :checked="item.check">
                </div>
              </div>
              <div class="col-10 text-left">
                {{ item.comorbidity }}
              </div>
            </div>
          </li>
          <li class="list-group-item list-group-item-dark"> Pulmonary Diseases</li>
          <li class="list-group-item list-group-item-action"
          v-for="(item, index) in pulmoDiseases"
          :key="item.comorbidity"
          v-on:click="aggregateConditions(pulmoDiseases, index)">
            <div class="row">
              <div class="col-2">
                <div class="form-check">
                  <input class="form-check-input" type="checkbox" :checked="item.check">
                </div>
              </div>
              <div class="col-10 text-left">
                {{ item.comorbidity }}
              </div>
            </div>
          </li>
          <li class="list-group-item list-group-item-dark"> Other Diseases</li>
          <li class="list-group-item list-group-item-action"
          v-for="(item, index) in otherDiseases"
          :key="item.comorbidity"
          v-on:click="aggregateConditions(otherDiseases, index)">
            <div class="row">
              <div class="col-2">
                <div class="form-check">
                  <input class="form-check-input" type="checkbox" :checked="item.check">
                </div>
              </div>
              <div class="col-10 text-left">
                {{ item.comorbidity }}
              </div>
            </div>
          </li>
          <li class="list-group-item list-group-item-dark"> Medications </li>
          <li class="list-group-item list-group-item-action"
          v-for="(item, index) in medications"
          :key="item.comorbidity"
          v-on:click="aggregateConditions(medications, index)">
            <div class="row">
              <div class="col-2">
                <div class="form-check">
                  <input class="form-check-input" type="checkbox" :checked="item.check">
                </div>
              </div>
              <div class="col-10 text-left">
                {{ item.comorbidity }}
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</template>

<script>
import { Condition } from '@/constants';

export default {
  name: 'ComoListComponent',
  data() {
    return {
      // The array we are going to pass up to the parent component
      // this gets populated with the current selections
      result: {},
      cardioDiseases: [
        /* Cardiovascular Diseases */
        { comorbidity: Condition.AtrialFib, check: false },
        { comorbidity: Condition.Defib, check: false },
        { comorbidity: Condition.CAD, check: false },
        { comorbidity: Condition.CardiacStent, check: false },
        { comorbidity: Condition.CerebralDisease, check: false },
        { comorbidity: Condition.PulmonaryVascular, check: false },
        { comorbidity: Condition.TIA, check: false },
        { comorbidity: Condition.Stroke, check: false },
        { comorbidity: Condition.VHD, check: false },
        { comorbidity: Condition.HeartFail, check: false },
        { comorbidity: Condition.PVD, check: false },
      ],
      pulmoDiseases: [
        /* Pulmonary Diseases */
        { comorbidity: Condition.PulmDisease, check: false },
      ],
      otherDiseases: [
        /* Other Diseases */
        { comorbidity: Condition.Bleeding, check: false },
        { comorbidity: Condition.Anemia, check: false },
        { comorbidity: Condition.ActiveBleeding, check: false },
        { comorbidity: Condition.Renal, check: false },
        { comorbidity: Condition.Diabetes, check: false },
        { comorbidity: Condition.Malignancy, check: false },
        { comorbidity: Condition.Hepatic, check: false },
        { comorbidity: Condition.Adrenal, check: false },
        { comorbidity: Condition.Pituitary, check: false },
        { comorbidity: Condition.Endocrine, check: false },
      ],
      medications: [
        /* Medications */
        { comorbidity: Condition.Digoxin, check: false },
        { comorbidity: Condition.Lithium, check: false },
        { comorbidity: Condition.Diuretics, check: false },
        { comorbidity: Condition.ACEI, check: false },
        { comorbidity: Condition.ARB, check: false },
        { comorbidity: Condition.NSAIDS, check: false },
        { comorbidity: Condition.Anticoagulant, check: false },
        { comorbidity: Condition.Antiplatelet, check: false },
        { comorbidity: Condition.Steroid, check: false },
      ],
    };
  },
  methods: {
    /**
    * Creates or removes a comorbidity to a String array to be
    * passed to the necessary functions. Executed whenever a
    * comorbidity is clicked in the list.
    * @returns comoList - List of comorbidities.
    */
    aggregateConditions: function aggregateConditions(comorbidityArray, index) {
      const array = comorbidityArray;

      // This method also updates the rendering
      // of the checkbox when, the list item is clicked.
      if (array[index].check === false) {
        array[index].check = true;
      } else {
        array[index].check = false;
      }

      this.$emit('updateEvent', { result: array[index].comorbidity });
    },
  },
};
</script>

<style scoped>
.half {
  height: 500px;
  overflow: scroll;
}
</style>
