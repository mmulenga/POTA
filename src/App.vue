<template>
  <div id="app">
    <DisclaimerModalComponent></DisclaimerModalComponent>
    <h1 class="my-4"> Pre-Op Testing App </h1>
      <div v-if="!isMobile() || (windowWidth > 767)" class="row">
        <!-- hide the status component when screen is smaller than md-->
        <div class="col-md-3 d-block">
            <StatusComponent
            :resultArray="resultArray"
            v-if="(windowWidth > 767)"/>
             <!-- This is the the bind to the child component-->
        </div>
        <!-- hide desktop como list when screen is smaller than md-->
        <div class="col-md-6 d-block">
            <ComoListComponent
            ref="ComoListComponent"
            v-if="(windowWidth > 767)"
            v-on:clickEvent="updateArray"
            v-on:hoverEvent="updateDescription"/>
            <ResultModalComponent
            ref="ResultModalComponent"
            v-if="(windowWidth > 767)"
            :resultArray="resultArray"
            v-on:reset-toggle="resetComoList"
            v-on:clear-results="clearResults"/>
        </div>
        <!-- hide desktop glossary when screen is smaller than md-->
        <div class="col-md-3 d-block">
            <GlossaryComponent
            :glossaryEntry="glossaryEntry"
            v-if="(windowWidth > 767)"/>
        </div>
    </div>
    <!-- visible-sm and down  (or hidden-md and up) -->
    <div v-if="isMobile() || (windowWidth <= 767)">
      <div>
      <!-- side drawer that contains the list of comos selected -->
        <drawer v-if="isMobile() || (windowWidth <= 767)"
        :show="drawerShow "
        v-on:on-hide="drawerToggle(), buttonsToggle(), resetScrollPosition()"
        v-on:submit-exams="submitExams">
          <div class="layout" slot="drawer" >
            <button id="drawer_close" type="button" class="close"
            v-on:click="drawerToggle(), buttonsToggle()">
                <span> &times; </span>
              </button>
            <!-- list component -->
            <StatusComponent
              :resultArray="resultArray"/>
          </div>
          <h1 class="my-4 bg-light" style="width: 100%"> Pre-Op Testing App </h1>
          <div class="col-md-12" style="overflow: auto">
            <MobileComoListComponent ref="MobileComoListComponent"
            v-on:clickEvent="updateArray"
            v-on:update-glossary="updateDescription"
            v-on:clear-glossary="clearDescription"
            v-on:toggle-buttons="buttonsToggle"
            v-on:reset-scroll-position="resetScrollPosition"/>
          </div>
          </drawer>
      </div>
      <ResultModalComponent
      v-if="isMobile() || (windowWidth <= 767)"
      ref="MobileResultModalComponent"
      class="navbar navbar-expand-lg navbar-light bg-light results"
      :hiddenButtons="buttonsHidden"
      :resultArray="resultArray"
      v-on:drawer-toggle="drawerToggle"
      v-on:hide-buttons="buttonsToggle"
      v-on:reset-toggle="resetMobileComoList"
      v-on:clear-results="clearResults"/>
  </div>
    </div>
</template>

<script>
import DisclaimerModalComponent from '@/components/DisclaimerModalComponent';
import ComoListComponent from '@/components/ComoListComponent';
import MobileComoListComponent from '@/components/MobileComoListComponent';
import GlossaryComponent from '@/components/GlossaryComponent';
import StatusComponent from '@/components/StatusComponent';
import ResultModalComponent from '@/components/ResultModalComponent';
import Drawer from '@/components/Drawer';

export default {
  name: 'App',
  components: {
    DisclaimerModalComponent,
    ComoListComponent,
    MobileComoListComponent,
    GlossaryComponent,
    StatusComponent,
    ResultModalComponent,
    Drawer,
  },
  data() {
    return {
      framework_name: 'VueJS',
      resultArray: [],
      glossaryEntry: '',
      drawerShow: false,
      buttonsHidden: false,
      windowWidth: 0,
    };
  },
  mounted() {
    this.getWindowWidth();
    // eslint-disable-next-line
    this.$nextTick(function () {
      window.addEventListener('resize', this.getWindowWidth);
      this.getWindowWidth();
    });
  },
  methods: {
    /**
     *
     */
    // eslint-disable-next-line
    getWindowWidth: function getWindowWidth(event) {
      this.windowWidth = document.documentElement.clientWidth;
      if (this.$refs.ResultModalComponent !== undefined) {
        this.$refs.ResultModalComponent.windowWidth = document.documentElement.clientWidth;
      } else {
        this.$refs.MobileResultModalComponent.windowWidth = document.documentElement.clientWidth;
      }
    },
    /**
    * Updates the resultArray used by the Patient Status window with data
    * recieved from child ComoListComponent.
    * @param comorbidity - The data emitted by the child component.
    */
    updateArray: function updateArray(comorbidity) {
      if (this.resultArray.includes(comorbidity.currentComorbiditySelection)) {
        // Apparently there's no delete by index, so this is the best we can do. :(
        this.resultArray.splice(this.resultArray.indexOf(comorbidity.currentComorbiditySelection),
          1);
      } else {
        this.resultArray.push(comorbidity.currentComorbiditySelection);
      }
    },
    /**
     * Updates the glossaryEntry used by the Glossary Window with data
     * recieved from child ComoListComponent.
     * @param comorbidity - The data emitted by the child component.
     */
    updateDescription: function updateDescription(comorbidity) {
      this.glossaryEntry = comorbidity.currentComorbiditySelection;
      if (this.$refs.MobileComoListComponent !== undefined) {
        this.$refs.MobileComoListComponent.currentComorbidityDescription
       = comorbidity.currentComorbiditySelection;
      }
    },
    /**
    * Clears the glossaryEntry used by the Glossary Window data
    * received from child ComoListComponent.
    */
    clearDescription: function clearDescription() {
      this.glossaryEntry = '';
    },

    /**
     * Displays and hides the drawer.
     */
    drawerToggle: function drawerToggle() {
      this.drawerShow = !this.drawerShow;
    },

    /**
     * Displays and hides the ResultsModalComponent
     * when the drawer or glossary is displayed.
     */
    buttonsToggle: function buttonsToggle() {
      this.buttonsHidden = !this.buttonsHidden;
    },

    /**
     * Resets the data of ComoListComponent.
     */
    resetComoList: function resetComoList() {
      this.$refs.ComoListComponent.resetData();
    },
    /**
     * Resets the data of MobileComoListComponent.
     */
    resetMobileComoList: function resetMobileComoList() {
      this.$refs.MobileComoListComponent.resetData();
    },
    /**
     * Sets the scroll position of the exam modal to 0,
     * which resets it to the top if user previously
     * scrolled down.
     */
    resetScrollPosition: function resetScrollPosition() {
      const tests = this.$el.querySelector('.show');
      if (tests !== null) {
        tests.scrollTop = 0;
      }
    },
    /**
    * This function is for the patient status drawer's submit button.
    * After recieving a "submit-exams" from the child, exams modal will
    * pop up showing all the required exams.
    */
    submitExams: function submitExams() {
      this.drawerToggle();
      this.buttonsToggle();
      if (this.$refs.ResultModalComponent !== undefined) {
        this.$refs.ResultModalComponent.getExams();
        this.$refs.ResultModalComponent.showModal();
      } else {
        this.$refs.MobileResultModalComponent.getExams();
        this.$refs.MobileResultModalComponent.showModal();
      }
    },
    /**
     * Clears the resultArray when the reset button is pressed.
     */
    clearResults: function clearResults() {
      this.resultArray = [];
    },
    /**
     * Checks to see if the user is viewing on a mobile device.
     */
    isMobile: function isMobile() {
      return (navigator.userAgent.indexOf('Mobile') !== -1);
    },
  },
};
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
}

.my-4 {
  margin-bottom: 0px !important;
  margin-top: 0px !important;
}
.layout {
  width: 300px;
}
.results {
  position: fixed;
}

button.close {
  padding-right: 10px;
  font-size: 50px;
}

</style>
