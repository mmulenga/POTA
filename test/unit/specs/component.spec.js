import Vue from 'vue';
// import Vue from 'vue/dist/vue';
import App from '@/App';
import Prototype from '@/components/Prototype';
import ComoListComponent from '@/components/ComoListComponent';
import { Condition, Exam } from '@/constants';

describe('Testing all comorbidities in ComoListComponent.vue', () => {
  it('returns correct list of cardiovascular diseases', () => {
    const Component = Vue.extend(ComoListComponent);
    const vm = new Component().$mount();
    const cardiovascularList = [{ comorbidity: Condition.AtrialFib },
      { comorbidity: Condition.Defib },
      { comorbidity: Condition.CAD },
      { comorbidity: Condition.CardiacStent },
      { comorbidity: Condition.CerebralDisease },
      { comorbidity: Condition.PulmonaryVascular },
      { comorbidity: Condition.TIA },
      { comorbidity: Condition.Stroke },
      { comorbidity: Condition.VHD },
      { comorbidity: Condition.HeartFail },
      { comorbidity: Condition.PVD }];
    expect(vm.cardioDiseases).toEqual(cardiovascularList);
  });

  it('returns correct list of pulmonary diseases', () => {
    const Component = Vue.extend(ComoListComponent);
    const vm = new Component().$mount();
    const pulmonaryList = [{ comorbidity: Condition.PulmDisease }];
    expect(vm.pulmoDiseases).toEqual(pulmonaryList);
  });

  it('returns correct list of other diseases', () => {
    const Component = Vue.extend(ComoListComponent);
    const vm = new Component().$mount();
    const otherDiseaseList = [{ comorbidity: Condition.Bleeding },
      { comorbidity: Condition.Anemia },
      { comorbidity: Condition.ActiveBleeding },
      { comorbidity: Condition.Renal },
      { comorbidity: Condition.Diabetes },
      { comorbidity: Condition.Malignancy },
      { comorbidity: Condition.Hepatic },
      { comorbidity: Condition.Adrenal },
      { comorbidity: Condition.Pituitary },
      { comorbidity: Condition.Endocrine }];
    expect(vm.otherDiseases).toEqual(otherDiseaseList);
  });

  it('returns correct list of medications', () => {
    const Component = Vue.extend(ComoListComponent);
    const vm = new Component().$mount();
    const medicationList = [{ comorbidity: Condition.Digoxin },
      { comorbidity: Condition.Lithium },
      { comorbidity: Condition.Diuretics },
      { comorbidity: Condition.ACEI },
      { comorbidity: Condition.ARB },
      { comorbidity: Condition.NSAIDS },
      { comorbidity: Condition.Anticoagulant },
      { comorbidity: Condition.Antiplatelet },
      { comorbidity: Condition.Steroid }];
    expect(vm.medications).toEqual(medicationList);
  });
});

describe('Testing framework in Prototype.vue', () => {
  it('returns correct framework being used', () => {
    const Component = Vue.extend(Prototype);
    const vm = new Component().$mount();
    expect(vm.framework_name).toEqual('VueJS');
  });
});

