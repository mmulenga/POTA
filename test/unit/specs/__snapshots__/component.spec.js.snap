// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing all comorbidities in ComoListComponent.vue List of cardiovascular diseases matches snapshot 1`] = `
Array [
  Object {
    "check": false,
    "comorbidity": "Atrial fibrillation / History of irregular heart beat",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Defibrillator / Pacemaker",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Coronary artery disease",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Cardiac Stent (< 12 months)",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "History of cerebral disease",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "History of pulmonary vascular disease",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Transient Ischemic Attack (TIA)",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Stroke",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Valvular heart disease / Valve Replacement",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Heart Failure",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Peripheral Vascular disease (PVD)",
    "glossary": "No extra information.",
  },
]
`;

exports[`Testing all comorbidities in ComoListComponent.vue List of medications matches snapshot 1`] = `
Array [
  Object {
    "check": false,
    "comorbidity": "Digoxin",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Lithium",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Diuretics",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "ACE-I",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "ARB",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "NSAIDS",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Anticoagulants",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Antiplatelet (ASA excluded)",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Systemic steroid use within 6 months",
    "glossary": "No extra information.",
  },
]
`;

exports[`Testing all comorbidities in ComoListComponent.vue List of other diseases matches snapshot 1`] = `
Array [
  Object {
    "check": false,
    "comorbidity": "Age > 69 years old",
    "glossary": undefined,
  },
  Object {
    "check": false,
    "comorbidity": ">= 2 risk factors (HTN, Proven OSA, or STOP-BANG >= 5, BMI >= 40",
    "glossary": undefined,
  },
  Object {
    "check": false,
    "comorbidity": "Bleeding disorders",
    "glossary": "• History of extensive investigation due to a bleeding episode. 
• History of extensive bleeding needing transfusion in previous surgeries. 
• Bruising easily more than usual. 
",
  },
  Object {
    "check": false,
    "comorbidity": "History of anemia",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "History of active bleeding",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Chronic kidney disease (eGFR < 60ml/min/1.73m^2)",
    "glossary": undefined,
  },
  Object {
    "check": false,
    "comorbidity": "Diabetes (on Insulin or 2 oral agents)",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Present malignancy / surgery for malignancy",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "Hepatic disease",
    "glossary": "• Alcohol abuse (i.e., female: 14 drinks / week; male: 21 drinks / week). 
• Jaundice due to liver disease. 
• All forms of hepatitis. 
• Cirrhosis; Hepatic cancer. 
",
  },
  Object {
    "check": false,
    "comorbidity": "History of adrenal disease",
    "glossary": "No extra information.",
  },
  Object {
    "check": false,
    "comorbidity": "History of pituitary disease",
    "glossary": "• Acromegaly. 
• Cushing’s disease. 
• Sheehan syndrome. 
• Growth hormone deficiency. 
• Hyper/hypo pituitarism. 
",
  },
  Object {
    "check": false,
    "comorbidity": "History of major systemic endocrine disease",
    "glossary": "(Excluding Diabeties) 
• Parathyroid disease 
• Addison’s disease 
• Cushing’s syndrome 
• Graves’ disease 
",
  },
  Object {
    "check": false,
    "comorbidity": "Thyroid disease or on thyroid supplementation",
    "glossary": undefined,
  },
]
`;

exports[`Testing all comorbidities in ComoListComponent.vue List of pulmonary diseases matches snapshot 1`] = `
Array [
  Object {
    "check": false,
    "comorbidity": "Severe COPD, Home Oxygen, Pulmonary HTN",
    "glossary": "• Spirometric values AND symptomatic assessment should be sought. 
• Post Bronchodilator FEV1: 30 - 50% of predicted; FEV1/FVC < 0.70 
• History of exacerbations (including prior hospitalizations) should be recorded.
• Worsening air flow limitation, greater SOB, reduced exercise capacity, having an impact on quality of life, and ability to complete activities of daily living should be present. 
",
  },
]
`;
